brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java

```diff
@@ -544,41 +546,98 @@ public void buildCopyOriginalFiles(File appDir)
+            ) {
+                byte[] buffer = new byte[4096 * 1024];
+                copyExistingFiles(inputFile, actualOutput, buffer);
+                copyUnknownFiles(appDir, actualOutput, files, buffer);
             } catch (IOException ex) {
                 throw new AndrolibException(ex);
             }
+
+            // Remove our temporary file.
+            tempFile.delete();
+        }
+    }
+
+    private void copyExistingFiles(ZipFile inputFile, ZipOutputStream outputFile, byte[] buffer) throws IOException {
+        // First, copy the contents from the existing outFile:
+        Enumeration<? extends ZipEntry> entries = inputFile.entries();
+        while (entries.hasMoreElements()) {
+            ZipEntry entry = new ZipEntry(entries.nextElement());
+            // We can't reuse the compressed size because it depends on compression sizes.
+            entry.setCompressedSize(-1);
+            outputFile.putNextEntry(entry);
+
+            // No need to create directory entries in the final apk
+            if (!entry.isDirectory()) {
+                copy(inputFile.getInputStream(entry), outputFile, buffer);
+            }
+
+            outputFile.closeEntry();
+        }
+    }
+
+    private void copyUnknownFiles(File appDir, ZipOutputStream outputFile, Map<String, String> files, byte[] buffer)
+            throws IOException {
+        File unknownFileDir = new File(appDir, UNK_DIRNAME);
+
+        // loop through unknown files
+        for (Map.Entry<String,String> unknownFileInfo : files.entrySet()) {
+            File inputFile = new File(unknownFileDir, unknownFileInfo.getKey());
+            if(inputFile.isDirectory()) {
+                continue;
+            }
+
+            ZipEntry newEntry = new ZipEntry(unknownFileInfo.getKey());
+            int method = Integer.valueOf(unknownFileInfo.getValue());
+            LOGGER.fine(String.format("Copying unknown file %s with method %d", unknownFileInfo.getKey(), method));
+            if(method == ZipEntry.STORED) {
+                newEntry.setMethod(ZipEntry.STORED);
+                newEntry.setSize(inputFile.length());
+                newEntry.setCompressedSize(-1);
+                BufferedInputStream unknownFile = new BufferedInputStream(new FileInputStream(inputFile));
+                CRC32 crc = calculateCrc(unknownFile, buffer);
+                newEntry.setCrc(crc.getValue());
+
+                LOGGER.fine("\tsize: " + newEntry.getSize());
+            }
+            else {
+                newEntry.setMethod(ZipEntry.DEFLATED);
+            }
+            outputFile.putNextEntry(newEntry);
+
+            BufferedInputStream unknownFile = new BufferedInputStream(new FileInputStream(inputFile));
+            copy(unknownFile, outputFile, buffer);
+            outputFile.closeEntry();
+        }
+    }
+
+    private CRC32 calculateCrc(InputStream input, byte[] buffer) throws IOException {
```

Comment:
Can you move these two functions (`calculateCrc` and `copy`) to `brut.j.util/src/main/java/brut/util/BrutIO.java` ?

Trying to prevent `Androlib` from growing too big, thus only containing functions directly related to Android & its resources.